### ISSUE TITLE
win_chocolatey error: "The package was not found with the source listed"

### ISSUE THREAD

**Issue by sparkvilla** [ansible/ansible-modules-extras#1886]
##### Issue Type:
- Feature Idea
##### Component Name:

win_chocolatey module
##### Ansible Version:

```
ansible 2.0.1.0
```
##### Ansible Configuration:

My `ansible.cfg` file:
    [defaults]
    hostfile = ./hosts

My `hosts` file:
    [win_clones]
    windows10 ansible_host=172.18.4.16 ansible_password="pass"

My `group_vars/win_clones` file: 
    ansible_user: diego
    ansible_port: 5986
    ansible_connection: winrm
    ansible_winrm_server_cert_validation: ignore 

My `choco_test.yml` playbook:
    - name: Test
      hosts: windows10
      tasks:
       - name: Install/Uninstall
         win_chocolatey: 
           name: vlc
           state: present
##### Environment:

Host = ubuntu 15.10 VM
Client = windows10 VM
I use a shared folder on NAS at : \174.18.4.17\choco-repo\choco-repo\
##### Summary:

I would like to use win_chocolatey module to pull vlc package present on my shared folder using Ansible. I therefore run:
ansible-playbook choco_test.yml
##### Actual Results:

I get the following error:

```
fatal: [windows10]: FAILED! => {"changed": false, "choco_error_cmd": "choco.exe install -dv -y vlc     -source \\\\172.18.4.15\\choco-repo\\choco-repo\\", "choco_error_log": "Chocolatey is running   on Windows v 10.0.10586.0 Attempting to delete file \"C:/ProgramData/chocolatey/choco.exe.old\". Attempting to delete file \"C:\\ProgramData\\chocolatey\\choco.exe.old\". Command line: \"C:\\ProgramData\\chocolatey\\choco.exe\" install -dv -y vlc -source \\\\172.18.4.15\\choco-repo\\choco-repo\\ Received arguments: install -dv -y vlc -source \\\\172.18.4.15\\choco-repo\\choco-repo\\  NOTE: Hiding sensitive configuration data! Please double and triple   check to be sure no  sensitive data is shown, especially if copying   output to a gist for review. Configuration: CommandName='install'| CacheLocation='C:\\Users\\diego\\AppData\\Local\\Temp'| ContainsLegacyPackageInstalls='True'| CommandExecutionTimeoutSeconds='2700'| Sources='\\\\172.18.4.15\\choco-repo\\choco-repo\\'|Debug='True'| Verbose='True'|Force='False'|Noop='False'|HelpRequested='False'| RegularOutput='True'|QuietOutput='False'|PromptForConfirmation='False'| AcceptLicense='True'|AllowUnofficialBuild='False'|Input='vlc'| AllVersions='False'|SkipPackageInstallProvider='False'| PackageNames='vlc'|Prerelease='False'|ForceX86='False'| OverrideArguments='False'|NotSilent='False'|IgnoreDependencies='False'| AllowMultipleVersions='False'|AllowDowngrade='False'| ForceDependencies='False'|Information.PlatformType='Windows'| Information.PlatformVersion='10.0.10586.0'| Information.PlatformName='Windows'| Information.ChocolateyVersion='0.9.9.11'| Information.ChocolateyProductVersion='0.9.9.11'| Information.FullName='choco, Version=0.9.9.11, Culture=neutral, PublicKeyToken=79d02ea9cad655eb'|  Information.Is64Bit='True'|Information.IsInteractive='False'| Information.IsUserAdministrator='True'| Information.IsProcessElevated='True'|Features.AutoUninstaller='True'| Features.CheckSumFiles='True'|Features.FailOnAutoUninstaller='False'| ListCommand.LocalOnly='False'| ListCommand.IncludeRegistryPrograms='False'| UpgradeCommand.FailOnUnfound='False'| UpgradeCommand.FailOnNotInstalled='False'| UpgradeCommand.NotifyOnlyAvailableUpgrades='False'| NewCommand.AutomaticPackage='False'|SourceCommand.Command='unknown'| SourceCommand.Priority='0'|FeatureCommand.Command='unknown'| ConfigCommand.Command='unknown'|PushCommand.TimeoutInSeconds='0'| PinCommand.Command='unknown'| _ Chocolatey:ChocolateyInstallCommand - Normal Run Mode _ Installing the following packages: vlc By installing you accept licenses for the packages. vlc not installed. The package was not found with the source(s) listed.  If you specified a particular version and are receiving this message, it is possible that the package name exists but the version does not.  Version: \"\"  Source(s): \"\\\\172.18.4.15\\choco-repo\\choco-repo\\\"  Chocolatey installed 0/1 package(s). 1 package(s) failed.  See the log for details (C:\\ProgramData\\chocolatey\\logs\\chocolatey.log). Failures:  - vlc Exiting with 1", "failed": true, "msg": "Error installing vlc"}
```

NOTE: I can pull vlc package from the shared folder using chocolatey directly from the client by running:

```
choco install vlc -y
```

Any help?

**Comment by jhawkesworth** [comment]
I am not a chocolatey user so I am guessing here.

Is there anything in C:\Program Data\chocolatey\logs\chocolatey.log ?

It is possible you are hitting a 'second hop' issue where fetching from the share is not allowed as it is a second hop to the share from the play host (from ansible controller to the play host would be the first hop).

I am not sure if it is possible but if you could try putting the source locally on the same machine where you want to install vlc then and see if that works, then you have at least narrowed down what the problem is.

HTH

Jon

**Comment by sparkvilla** [comment]
Hi Jon. Thanks for your replay. I tried to put the source locally and It does work! It seems that I am hitting the 'second hop' issue... The output of the log file resulting from the second hop issue:

```
Information.Is64Bit='True'|Information.IsInteractive='False'|
Information.IsUserAdministrator='True'|
Information.IsProcessElevated='True'|Features.AutoUninstaller='True'|
Features.CheckSumFiles='True'|Features.FailOnAutoUninstaller='False'|
ListCommand.LocalOnly='False'|
ListCommand.IncludeRegistryPrograms='False'|
UpgradeCommand.FailOnUnfound='False'|
UpgradeCommand.FailOnNotInstalled='False'|
UpgradeCommand.NotifyOnlyAvailableUpgrades='False'|
NewCommand.AutomaticPackage='False'|SourceCommand.Command='unknown'|
SourceCommand.Priority='0'|FeatureCommand.Command='unknown'|
ConfigCommand.Command='unknown'|PushCommand.TimeoutInSeconds='0'|
PinCommand.Command='unknown'|
2016-03-23 11:33:31,808 [DEBUG] - _ Chocolatey:ChocolateyInstallCommand - Normal Run   Mode _
2016-03-23 11:33:31,824 [INFO ] - Installing the following packages:
2016-03-23 11:33:31,824 [INFO ] - vlc
2016-03-23 11:33:31,824 [INFO ] - By installing you accept licenses for the packages.
2016-03-23 11:33:31,918 [ERROR] - vlc not installed. The package was not found with the    source(s) listed.
 If you specified a particular version and are receiving this message, it is possible that the package name exists but the version does not.
 Version: ""
 Source(s): "\\172.18.4.15\choco-repo\choco-repo\"
2016-03-23 11:33:31,918 [WARN ] - 
Chocolatey installed 0/1 package(s). 1 package(s) failed.
See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
2016-03-23 11:33:31,933 [ERROR] - Failures:
2016-03-23 11:33:31,933 [ERROR] -  - vlc
2016-03-23 11:33:31,933 [DEBUG] - Exiting with 1
```

**Comment by jhawkesworth** [comment]
My guess would be if you looked in the C:\ProgramData\chocolatey\logs\chocolatey.log
there would be some kind of message saying something along the lines of 'access denied'.

Right now, I don't believe there is a direct solution for the second hop issue, other than working around it (by using a local source, for example).  My understanding is that it is likely to need CredSSP support adding to the winrm connection plugin.

Although I don't use chocolatey, mostly I do put the software components I need to install onto windows hosts before running the installers so if you can use a local source it is probably a reasonably workaround.

HTH

Jon

**Comment by nitzmahone** [comment]
Suspect the only way to make this work would be to add username/password share access support to the chocolatey module (ala what's done in win_package)- I'd say use kerberos delegation, but I'm guessing your NAS is just doing basic workgroup share emulation w/ username/password, so no dice there. Tagging this as a feature request.

**Comment by ansibot** [comment]
@smadam813, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by nitzmahone** [comment]
This would likely also be solved by most of the become stuff I've prototyped for 2.3 (you'd [re]specify network credentials for become that would create an interactive/network session), rather than adding that functionality into the module itself.

**Comment by ansibot** [comment]
@smadam813, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@smadam813, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by dagwieers** [comment]
@sparkvilla Please test your issue with the v2.3 release, or the devel branch. And report back :-) If this issue is still present, please move it to https://github.com/ansible/ansible

Thanks in advance !

**Comment by dagwieers** [comment]
A lot of the supporting code has been refactored and we need feedback from @sparkvilla.
If you still have an issue with v2.3 or the devel branch, please file a new issue at https://github.com/ansible/ansible/issues
